### PR TITLE
Fix to stop forwarding proposals in consumers after scaling down a stream

### PR DESCRIPTION
**Pull request by wallyqs**
Sometimes when scaling down a stream, a raft node could continue forwarding proposals after already being closed, in the debug logs this can be confirmed by many entries logging `Direct proposal ignored, not leader (state: CLOSED)`.

### CODE REVIEW

**Review by derekcollison — APPROVED** [review]
LGTM - How long does test take?

File: server/consumer.go [hunk 1]
```diff
@@ -1122,6 +1122,16 @@ func (o *consumer) isLeader() bool {
 	return true
 }
 
+func (o *consumer) clearLoopAndForward() {
+	o.mu.Lock()
+	defer o.mu.Unlock()
+	if o.qch != nil {
+		close(o.qch)
+		// Note can not close pch here.
```
Comment (neilalexander) [inline server/consumer.go:1130]: Is this because of a panic when writing to the channel elsewhere?

**Comment by wallyqs** [comment]
Currently it takes about 40s, I think it can be made shorter

File: server/jetstream_cluster.go [hunk 2]
```diff
@@ -4264,6 +4264,8 @@ func (js *jetStream) processClusterCreateConsumer(ca *consumerAssignment, state
 		} else {
 			// Check for scale down to 1..
 			if rg.node != nil && len(rg.Peers) == 1 {
+				// Need to pop loopAndForward by closing qch and nil out both qch and pch.
```
Comment (neilalexander) [inline server/jetstream_cluster.go:4267]: Would be nice if we can expand this comment just to explain what happens if we don't, i.e. that it results in errant proposals.

**Review by neilalexander — APPROVED** [review]
Otherwise LGTM

File: server/jetstream_cluster.go (hunk 2 above)
Comment (wallyqs) [reply server/jetstream_cluster.go]: Updated this part

**Review by neilalexander — APPROVED** [review]
LGTM

**Pull request merged**